src/main/java/io/appium/java_client/pagefactory/DefaultElementByBuilder.java

```diff
@@ -119,105 +112,81 @@ private static By buildMobileBy(LocatorGroupStrategy locatorGroupStrategy, Annot
+            Method value;
+            Annotation[] set;
+            try {
+                value = annotationClass.getMethod(VALUE, ANNOTATION_ARGUMENTS);
+                set = (Annotation[]) value.invoke(a, ANNOTATION_PARAMETERS);
+            } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+                throw new ClassCastException(String.format("The annotation '%s' has no convenient '%s' method which "
+                        + "returns array of annotations", annotationClass.getName(), VALUE));
             }
 
-            if (androidFindBys != null) {
-                return createBy(androidFindBys.value(), HowToUseSelectors.BUILD_CHAINED);
+            sort(set, comparator);
+            if (chainedLocator.equals(annotationClass)) {
+                result = add(result, createBy(set, HowToUseSelectors.BUILD_CHAINED));
+                continue;
             }
 
-            if (androidFindAll != null) {
-                return createBy(androidFindAll.value(), HowToUseSelectors.USE_ANY);
-            }
-            ///////////////////////////////////////
-            //code that supposed to be supported
-            if (androidFindByArray != null && androidFindByArray.length > 0) {
-                return buildMobileBy(howToUseLocators != null ? howToUseLocators.androidAutomation() : null,
-                        androidFindByArray);
+            if (allLocator.equals(annotationClass)) {
+                result = add(result, createBy(set, HowToUseSelectors.USE_ANY));
             }
         }
 
-        if (isIOSXcuit()) {
-            iOSXCUITFindBy[] xCuitFindByArray = annotatedElement.getAnnotationsByType(iOSXCUITFindBy.class);
-            if (xCuitFindByArray != null && xCuitFindByArray.length > 0) {
-                return buildMobileBy(howToUseLocators != null ? howToUseLocators.iOSXCUITAutomation() : null,
-                        xCuitFindByArray);
-            }
-        }
+        return result;
+    }
 
-        if (isIOS()) {
-            iOSFindBy[] iOSFindByArray = annotatedElement.getAnnotationsByType(iOSFindBy.class);
-            //should be kept for some time
-            iOSFindBys iOSFindBys = annotatedElement.getAnnotation(iOSFindBys.class);
-            iOSFindAll iOSFindAll = annotatedElement.getAnnotation(iOSFindAll.class);
+    @Override protected By buildMobileNativeBy() {
+        AnnotatedElement annotatedElement = annotatedElementContainer.getAnnotated();
+        HowToUseLocators howToUseLocators = annotatedElement.getAnnotation(HowToUseLocators.class);
 
-            if (iOSFindByArray != null && iOSFindByArray.length == 1) {
-                return createBy(new Annotation[] {iOSFindByArray[0]}, HowToUseSelectors.USE_ONE);
-            }
+        By result = null;
+        if (isSelendroidAutomation()) {
+            result =  buildMobileBy(howToUseLocators != null ? howToUseLocators.selendroidAutomation() : null,
```

Comment:
I'm always a bit suspicious about using method calls with ternary operator, since Java will invoke the method even if the precondition equals to false